homeassistant/components/emulated_hue/hue_api.py

```diff
@@ -556,16 +591,27 @@ async def put(self, request, username, entity_number):
                     create_hue_success_response(entity_number, val, parsed[key])
                 )
 
-        # Echo fetches the state immediately after the PUT method returns.
-        # Waiting for a short time allows the changes to propagate.
-        await asyncio.sleep(0.25)
-
         return self.json(json_response)
 
 
 def get_entity_state(config, entity):
     """Retrieve and convert state and brightness values for an entity."""
-    cached_state = config.cached_states.get(entity.entity_id, None)
+    cached_state_entry = config.cached_states.get(entity.entity_id, None)
+    cached_state = None
+
+    # Check if we have a cached entry, and if so if it hasn't expired.
+    if cached_state_entry is not None:
+        if cached_state_entry[STATE_CACHE_TIME] is None:
+            # Handle the where the entity is listed in config.off_maps_to_on_domains.
+            cached_state = cached_state_entry[STATE_CACHE_STATE]
+        elif time.time() - cached_state_entry[STATE_CACHE_TIME] < STATE_CACHED_TIMEOUT:
+            # We do want to report the actual state of the entity.
+            cached_state = cached_state_entry[STATE_CACHE_STATE]
+            cached_state[STATE_ON] = entity.state != STATE_OFF
```

Comment:
Then we need a new test for the caching

Reply:
Is there more to do than the changes you've already made?

what would you like to test?

Reply:
A test to make sure that the GET response is the same as the PUT response within 2 seconds if the service call is successful and the state changes.

A test to make sure that the GET response does not use the cache if PUT response within 2 seconds if the service call is Unsuccessful and the state does not change.